### ISSUE TITLE
[FEATURE] Use actual password validation message as help text

### ISSUE THREAD

**Issue by AlexRuiz7** [opensearch-project/security-dashboards-plugin#1501]
## Description

The help text used for the inputs in the `Create internal user` is hard-coded, so it doesn't match with the actual requirements if these are changes using the proper options in the backend plugin:

```
plugins.security.restapi.password_validation_regex: '(?=.*[A-Z])(?=.*[^a-zA-Z\d])(?=.*[0-9])(?=.*[a-z]).{5,}'
plugins.security.restapi.password_validation_error_message: "Password must be minimum 5 characters long and must contain at least one uppercase letter, one lowercase letter, one digit, and one special character."
```

I assume that this is happening for other inputs as well, not only for the password one.

It would be great if the `plugins.security.restapi.password_validation_error_message` string were used for the help text, instead of the hard-coded string.

https://github.com/opensearch-project/security-dashboards-plugin/blob/32c02c8c18a509884613517fac38b684338d5273/public/apps/apps-constants.tsx#L20-L22

![image](https://github.com/opensearch-project/security-dashboards-plugin/assets/15186973/d452a39d-d622-4aa8-b869-397233030057)

**Comment by scrawfor99** [comment]
[Triage] This issue will be addressed with the completion of #1503.

**Comment by davidlago** [comment]
Included in 2.x/2.9 branches
